### PR TITLE
feat(accuracy): MATH-500 lighteval-backed benchmark (AIP-879)

### PR DESCRIPTION
**Pull request by debermudez**
- **Plugin**: registers `math_500` benchmark in `plugins.yaml` (`default_grader: math`, `default_n_shots: 0`); scaffold loader raises `NotImplementedError` until the full lighteval-backed implementation lands.
- 500 curated math problems spanning algebra, geometry, number theory, and combinatorics; gold answers are LaTeX snippets (e.g. `\frac{1}{3}`, `\sqrt{2}`).
- Uses `LightevalLatexGrader` (`lighteval_latex`, `latex_gold_metric`, `LatexExtractionConfig` on the gold side) introduced in AIP-874.
- Depends on AIP-876 (lighteval sub-stack ordering).

Reference: `trt-llm-benchmark-recipe/src/accuracy/acc_bench_lighteval.py` (`math_500` task)

## Summary by CodeRabbit

* **New Features**
  * MATH-500 benchmark implemented and available for accuracy testing with lighteval LaTeX grading and boxed-expression extraction.

* **Documentation**
  * Accuracy docs updated to mark MATH-500 implemented and adjust benchmark status (remaining stubs reduced to two).

* **Tests**
  * Added unit tests covering prompts, ground truth, task mapping, metadata, and dataset edge cases for MATH-500.

### CODE REVIEW

**Comment by debermudez** [comment]
## Stack dependency

This PR is part of an 8-PR stack aligning aiperf's accuracy benchmarks with
the trt-llm-benchmark-recipe reference implementation. The branches were
rebased together; each PR depends on its parent landing first.

**Merge order:**

1. AIP-874 — #849  ← foundation (base: `main`)
2. AIP-877 — #923  (DeepEval stack)
3. AIP-878 — #924
4. AIP-875 — #925  (lighteval stack)
5. AIP-876 — #926
6. AIP-879 — #927
7. AIP-880 — #928
8. AIP-881 — #929

**This PR:** position **6 of 8** — base branch is `dbermudez/aip-876-implement-aime25-benchmark-loader`,
depends on **#926** (AIP-876) landing first.

After each upstream PR merges, the downstream PR's branch will be rebased
onto the updated parent before its own merge.

File: src/aiperf/accuracy/benchmarks/math_500.py [hunk 1]
```diff
@@ -1,31 +1,117 @@
...
+            One ``BenchmarkProblem`` per dataset row, in dataset order.
+        """
+        ds: Dataset = await asyncio.to_thread(load_dataset, DATASET_NAME, split="test")
+        return await asyncio.to_thread(self._build_problems, ds)
+
+    def _build_problems(self, ds: Dataset) -> list[BenchmarkProblem]:
+        problems: list[BenchmarkProblem] = []
+        for row in ds:
+            problem = row[PROBLEM_FIELD]
+            solution = str(row.get(SOLUTION_FIELD, ""))
```
[the remark this answers is not in the document]
Comment (debermudez) [reply src/aiperf/accuracy/benchmarks/math_500.py]: addressed

File: tests/unit/accuracy/test_math_500_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+
+def _make_run():
+    return make_benchmark_run(
+        model_names=["test-model"],
+        endpoint_type=EndpointType.COMPLETIONS,
+        streaming=False,
+        accuracy={"benchmark": AccuracyBenchmarkType.MATH_500},
+    )
+
```
[the remark this answers is not in the document]
Comment (debermudez) [reply tests/unit/accuracy/test_math_500_benchmark.py]: addressed

**Review by FrankD412 — APPROVED** [review]
